pydy/system.py

```diff
@@ -340,22 +341,25 @@ def times(self):
         equations of motion are integrated, numerically.
 
         The object should be in a format which the integration module to be
-        used can accept. Since this attribute is not checked for
-        compatibility, the user becomes responsible to supply it correctly.
+        used can accept.
         """
         return self._times
 
     @times.setter
     def times(self, new_times):
-        self._times = new_times
+        self._times = np.asarray(new_times)
+        self._check_times(self._times)
 
     def _check_times(self, times):
-        """
-        Very basic checking.
-        TODO: add more checking
-        """
-        if len(times) == 0:
+        if len(times.shape) == 0:
```

Comment:
This just checks to make sure the array has at least 1 dimension. This doesn't tell you if this is an array or not. The previous check made sure that the length of the array was greater than zero.

`asarray` will throw an error if what you pass in isn't array like. So this check doesn't make any sense. What are you trying to check?